torch/csrc/lazy/core/ir.h

```diff
@@ -180,6 +182,9 @@ class TORCH_API Node {
   // Outputs do not hold references on the nodes, and neither do the uses, since
   // otherwise we get into circular reference counting.
   std::vector<Output> operands_as_outputs_;
+  // In order to make copies of IR nodes, we need to access the operands in the
+  // format which is passed into the constructor.
+  OpList operands_as_oplist_;
```

Comment:
Instead of adding a data member just for clone, maybe we should add a protected ctor that takes all the data members as the parameters?

Reply:
Also, with operands_as_oplist_, we now have three data member to represent operands. It's a totally mess. @steventk-g Do you think you could come up with a solution that we can just use one data member? If not, I guess it's okay to leave it as it is and leave a TODO here.